src/base/server_type.cpp

```diff
@@ -45,16 +45,18 @@ SERVER_TYPE get_server_type ()
 
 #if defined (SERVER_MODE)
 
-void init_server_type (const char *db_name)
+int init_server_type (const char *db_name)
 {
+  int er_code = NO_ERROR;
   g_server_type = (SERVER_TYPE) prm_get_integer_value (PRM_ID_SERVER_TYPE);
   if (g_server_type == SERVER_TYPE_TRANSACTION)
     {
-      ats_Gl.init_page_server_hosts (db_name);
+      er_code = ats_Gl.init_page_server_hosts (db_name);
     }
 
   er_log_debug (ARG_FILE_LINE, "Starting server type: %s\n",
 		get_server_type () == SERVER_TYPE_PAGE ? "page" : "transaction");
+  return er_code;
```

Comment:
- ASSERT_ERROR check guarantees that er_set was called and thus problems will be easier to track.
- "Starting server type" message should not be printed if an error occurred.
```suggestion
      er_code = ats_Gl.init_page_server_hosts (db_name);
      if (er_code != NO_ERROR)
        {
          ASSERT_ERROR ();
          return er_code;
        }
    }

  er_log_debug (ARG_FILE_LINE, "Starting server type: %s\n",
		get_server_type () == SERVER_TYPE_PAGE ? "page" : "transaction");
  return NO_ERROR;
```

Reply:
Done